tests/test_docker_publish_workflow.py

```
@@ -0,0 +1,36 @@
+
+    def test_workflow_configures_qemu_before_buildx_for_multi_arch_images(self) -> None:
+        qemu_setup = (
+            "        uses: docker/setup-qemu-action@"
+            "68827325e0b33c7199eb31dd4e31fbe9023e06e8 # v3.0.0"
+        )
+        buildx_setup = (
+            "        uses: docker/setup-buildx-action@"
+            "f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0"
+        )
+
+        self.assertIn(qemu_setup, self.lines)
```

Ruff (0.15.5)

[warning] Use a regular `assert` instead of unittest-style `assertIn`

Replace `assertIn(...)` with `assert ...`

(PT009)